File: guests/go/select/select.go
// The select conformance scene, Go port. See
// guests/rust/select.rs and tools/scenes/select.steps.
package selectscene // `select` is a keyword; C# spells it this way too

import (
	kaya "dev.kaya/bindings/go"
)

var options = []string{"Red", "Green", "Blue"}

func App() *kaya.App {
	app := kaya.NewApp()

	var picked kaya.Signal[string]
	app.Build(func(tx *kaya.Tx) {
		tx.Window(0).Title("select")
		picked = tx.Signal("picked: Red")

		tx.Mount(tx.Column(func() {
			tx.Select(options, 0, func(tx *kaya.Tx, index int) {
				tx.Write(picked, "picked: "+options[index])
			})
			tx.Label(picked) // label#0
		}))
	})

	return app
}

File: guests/java/dev/kaya/guests/Select.java
package dev.kaya.guests;

import dev.kaya.KayaApp;

/**
 * The select scene from the JVM — guests/rust/select.rs,
 * tools/scenes/select.steps.
 */
public final class Select {
    private static final String[] OPTIONS = {"Red", "Green", "Blue"};

    public static void app() {
        KayaApp app = new KayaApp();

        app.build(tx -> {
            tx.window(0).title("select");
            KayaApp.Signal<String> picked = tx.signal("picked: Red");

            tx.mount(tx.column(() -> {
                tx.select(OPTIONS, 0, (t, index) ->
                        t.write(picked, "picked: " + OPTIONS[index]));
                tx.label(picked); // label#0
            }));
            return null;
        });

        app.dispatchLoop();
    }

    private Select() {}
}
